src/__test__/pages/experiments/[experimentId]/data-exploration/components/differential-expression-tool/DiffExprResults.test.jsx

```diff
@@ -214,4 +216,18 @@ describe('DiffExprResults', () => {
       lookupComponent.unmount();
     });
   });
+  it('Show comparison settings button works.', () => {
+    const component = mount(
+      <Provider store={store}>
+        <DiffExprResults experimentId='1234' onGoBack={jest.fn()} width={100} height={200} />
+      </Provider>,
+    );
+    const button = component.find(Button).at(1);
+    expect(button.childAt(0).text()).toEqual('Show settings');
+    button.simulate('click');
+    expect(button.childAt(0).text()).toEqual('Hide settings');
+
+    const div = component.find('div');
+    expect(div.at(0).childAt(1).text()).toEqual('condition-control vs. louvain-0 in louvain-1');
```

Comment:
this is not a good way to find the text for the chosen options, because if we changed the component and added an additional div, this test will break. I would suggest that in the `DiffExprResults` component, you give an id to the `div` that contains the data for selected options, and then in here you use a css selector to find the correct div using this id (for examples of how to use css selectors and find, see test `DiffExprResults.test.jsx` and the usages of component.find in there)